### PR TITLE
feat(server): authenticate user before ws connected

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7777** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7777?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/fundamentals/utils/request.ts [hunk 1]
```diff
@@ -66,3 +66,26 @@ export function getRequestFromHost(host: ArgumentsHost) {
...
+        cookies[decodeURIComponent(key.trim())] = val
+          ? decodeURIComponent(val.trim())
+          : val;
+      }
+
+      return cookies;
+    },
+    {} as Record<string, string>
+  );
+  req.cookies;
```
Comment (darkskygit) [inline packages/backend/server/src/fundamentals/utils/request.ts]: extra expression?

File: packages/backend/server/src/fundamentals/utils/request.ts [hunk 2]
```diff
@@ -66,3 +66,26 @@ export function getRequestFromHost(host: ArgumentsHost) {
 export function getRequestResponseFromContext(ctx: ExecutionContext) {
   return getRequestResponseFromHost(ctx);
 }
+
+export function parseCookies(req: Request) {
```
Comment (darkskygit) [inline packages/backend/server/src/fundamentals/utils/request.ts:74]: need some function comments to explain the behavior that will overwrite req.cookies

File: packages/backend/server/src/global.d.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 declare namespace Express {
   interface Request {
     user?: import('./core/auth/current-user').CurrentUser;
-    sid?: string;
+    session?: import('./core/auth/current-user').UserSessionl;
```
Comment (darkskygit) [inline packages/backend/server/src/global.d.ts]: typo

**Review by EYHN — APPROVED** [review]

**Pull request merged**